docs/Getting Started/Integration Basics/index.md

```diff
@@ -1236,3 +1236,127 @@ The `SDLSessionBean` should be inside a Java package other than the default pack
+
+##### Node.js TCP Transport
+```js
+lifecycleConfig.setTransportConfig(new SDL.transport.TcpClientConfig(HOST, PORT));
+```
+
+##### Node.js WebSocket Server
+```js
+lifecycleConfig.setTransportConfig(
+    new SDL.transport.WebSocketServerConfig(
+        wsClient, // The developer is responsible for providing incoming client connections
+        CONNECTION_LOST_TIMEOUT // connection timeout in milliseconds
+    )
+);
+```
+##### Vanilla JS WebSocket Client
+```js
+lifecycleConfig.setTransportConfig(new SDL.transport.WebSocketClientConfig(HOST, PORT));
+```
+
+## Additional Configuration Options
+There are several additional basic configuration options to set up your app, like the app name and icon.
+
+### App Icon
+An app icon can be set in the `LifecycleConfig` to automatically upload and set the icon image. Note that although the implementation of retreiving files are different between the JS browser and Node.js environments, the developer can use the same API in both cases, and the SDL library will cover the implementation details for the developer depending on which build they are using.
+
+```js
+const filePath = './app_icon.png';
+const file = new SDL.manager.file.filetypes.SdlFile()
+    .setName('AppIcon')
+    .setFilePath(filePath)
+    .setType(SDL.rpc.enums.FileType.GRAPHIC_PNG)
+    .setPersistent(true);
+
+lifecycleConfig.setAppIcon(file);
+```
+
+In this case, the code snippet expects there to be an `app_icon.png` file present in the same directory for the app icon.
+
+## Setting Up the SDL Manager
+After creating the `LifecycleConfig`, it can be set into the `AppConfig` and then passed into the `SdlManager`. The following snippet will set up the `SdlManager` and start it up. A listener is attached to the manager listener to let you know when there is a connection and the managers are ready. Listeners can also be set up for RPCs in the `SdlManager`. In the snippet below, the `OnHMIStatus` notification is subscribed to in order to know what HMI level the app is in.
+
+```js
+const appConfig = new SDL.manager.AppConfig()
+    .setLifecycleConfig(lifecycleConfig);
+
+const managerListener = new SDL.manager.SdlManagerListener()
+    .setOnStart((sdlManager) => {
+        // managers are ready
+    })
+    .setOnError((sdlManager, info) => {
+        console.error('Error from SdlManagerListener: ', info);
+    });
+
+const sdlManager = new SDL.manager.SdlManager(appConfig, managerListener)
+    .start()
+    .addRpcListener(SDL.rpc.enums.FunctionID.OnHMIStatus, (onHmiStatus) => {
+        // HMI Level updates
+        const hmiLevel = onHmiStatus.getHmiLevel();
+        console.log("Current HMI Level: ", hmiLevel);
+    });
+```
+
+For WebEngine apps, the resulting `index.html` may look something like this:
```

Comment:
I think this still needs a header like:

```md
## Configuring the WebEngine App HTML File
```

and a short intro. Starting with "the resulting `index.html` file..." is a little confusing.